src/pages/TokenDetails/Widget.tsx

```diff
@@ -0,0 +1,183 @@
+import { TransactionReceipt } from '@ethersproject/abstract-provider'
+import { Percent } from '@uniswap/sdk-core'
+import {
+  Currency,
+  Field,
+  Slippage,
+  SwapController,
+  SwapEventHandlers,
+  SwapSettingsController,
+  SwapWidget,
+  TradeType,
+} from '@uniswap/widgets'
+import { useWeb3React } from '@web3-react/core'
+import CurrencySearchModal from 'components/SearchModal/CurrencySearchModal'
+import { DEFAULT_DEADLINE_FROM_NOW } from 'constants/misc'
+import { RPC_URLS } from 'constants/networks'
+import { useActiveLocale } from 'hooks/useActiveLocale'
+import { useCallback, useMemo, useState } from 'react'
+import {
+  useIsDarkMode,
+  useSetUserSlippageTolerance,
+  useUserSlippageTolerance,
+  useUserTransactionTTL,
+} from 'state/user/hooks'
+import { DARK_THEME, LIGHT_THEME } from 'theme/widget'
+
+export const WIDGET_WIDTH = 320
+
+const WIDGET_ROUTER_URL = 'https://api.uniswap.org/v1/'
+
+export interface WidgetProps {
+  defaultToken: Currency
+}
+
+export default function Widget({ defaultToken }: WidgetProps) {
+  const locale = useActiveLocale()
+  const darkMode = useIsDarkMode()
+  const theme = useMemo(() => (darkMode ? DARK_THEME : LIGHT_THEME), [darkMode])
+  const { provider } = useWeb3React()
+
+  const { value, tokenSelector } = useSyncWidgetValue(defaultToken)
+  const { settings } = useSyncWidgetSettings()
+  const { transactions } = useSyncWidgetTransactions()
+
+  return (
+    <>
+      <SwapWidget
+        hideConnectionUI
+        jsonRpcUrlMap={RPC_URLS}
+        routerUrl={WIDGET_ROUTER_URL}
+        width={WIDGET_WIDTH}
+        locale={locale}
+        theme={theme}
+        provider={provider}
+        {...value}
+        {...settings}
+        {...transactions}
+      />
+      {tokenSelector}
+    </>
+  )
+}
+
+// Integrates the Widget's settings, keeping the widget and app settings in sync.
+function useSyncWidgetSettings() {
+  const [userTtl, setUserTtl] = useUserTransactionTTL()
+  const [ttl, setTtl] = useState<number | undefined>(userTtl / 60)
+  const onTransactionDeadlineChange = useCallback(
+    (ttl: number | undefined) => {
+      setTtl(ttl)
```

Comment:
why do ttl and userTtl need to be stored separately?

Reply:
they are different types, and different units, so they must be converted. this acts as the integration layer to eg allow undefined (defaulting) and convert between seconds/minutes.

Reply:
adding a note and prefixing the types to make this more apparent.